test/e2e/reloadinterval_test.go

```diff
@@ -0,0 +1,133 @@
+		{"set reloadInterval 1s", metav1.Duration{-1 * time.Second}, "5s", false},
+
+		// the following values all pass the regex validation
+		//
+		// set Spec.TuningOptions.ReloadInterval to a nondefault value within allowed bounds
+		// expect no error
+		{"set reloadInterval 100s", metav1.Duration{100 * time.Second}, "100s", true},
+		{"set reloadInterval 1m", metav1.Duration{1 * time.Minute}, "1m", true},
+
+		// set ReloadInterval to 0, which should give default of 5s
+		// expect no error
+		{"set reloadInterval default", metav1.Duration{0 * time.Second}, "5s", true},
+
+		// set ReloadInterval to a value outside of allowed bounds
+		// expect no error
+		{"set reloadInterval 130s", metav1.Duration{130 * time.Second}, "2m", true},
+	} {
+		t.Log(testCase.description)
+		// cases with values that pass the regex validation
+		if testCase.expectSuccess {
+			if err := setReloadIntervalValidValue(t, kclient, 1*time.Minute, testCase.reloadInterval, icName); err != nil {
+				t.Fatalf("failed to update ingresscontroller with reloadInterval=%v: %v", testCase.reloadInterval, err)
+			}
+			// cases with values that fail the regex validation
+		} else {
+			err := setReloadIntervalInvalidValue(t, kclient, testCase.reloadInterval, icName)
+			if err == nil {
+				t.Fatalf("expected an error")
+			}
+			if !strings.Contains(err.Error(), expectedErr) {
+				t.Fatalf("expected error message %q, got %v", expectedErr, err)
+			}
+		}
+
+		if err := waitForIngressControllerCondition(t, kclient, 5*time.Minute, icName, availableConditionsForPrivateIngressController...); err != nil {
+			t.Fatalf("failed to observe expected conditions: %v", err)
+		}
+		if err := waitForDeploymentEnvVar(t, kclient, deployment, 1*time.Minute, ingresscontroller.RouterReloadIntervalEnvName, testCase.expectedEnvVar); err != nil {
+			t.Fatalf("router deployment not updated with %s=%v: %v", ingresscontroller.RouterReloadIntervalEnvName, testCase.expectedEnvVar, err)
+		}
+	}
+}
+
+func setReloadIntervalValidValue(t *testing.T, client client.Client, timeout time.Duration, reloadInterval metav1.Duration, name types.NamespacedName) error {
+	t.Helper()
+
+	return wait.PollImmediate(1*time.Second, timeout, func() (bool, error) {
+		ic := operatorv1.IngressController{}
+		if err := client.Get(context.TODO(), name, &ic); err != nil {
+			t.Logf("Get %q failed: %v, retrying ...", name, err)
+			return false, nil
+		}
+		ic.Spec.TuningOptions.ReloadInterval = reloadInterval
+		if err := client.Update(context.TODO(), &ic); err != nil {
+			t.Logf("Update %q failed: %v retrying ...", name, err)
+			return false, nil
+		}
+		return true, nil
+	})
+}
+
+func setReloadIntervalInvalidValue(t *testing.T, client client.Client, reloadInterval metav1.Duration, name types.NamespacedName) error {
+	t.Helper()
+
+	ic := operatorv1.IngressController{}
+	if err := client.Get(context.TODO(), name, &ic); err != nil {
+		t.Logf("Get %q failed: %v, retrying ...", name, err)
```

Comment:
nit: This error message is not quite right, as it wouldn't be retrying here, since it is no longer wrapped in the `wait.PollImmediate` loop